apps/docs/registry.mdx

```diff
@@ -0,0 +1,235 @@
+```
+
+**Access levels:**
+- `public` – Anyone can run the codemod
+- `private` – Only the owner can run the codemod
+- `pro` – Only Pro plan users can run the codemod
+
+**Visibility levels:**
+- `public` – Everyone can see the package in search and listings
+- `private` – Only the owner can see the package
+- `org` – Members of the package's organization scope can see it
+- `user` – Visible only to the publishing user
+
+<Tip>
+  Access and visibility work together. For example, `visibility: public` with `access: pro` shows the package publicly, but only Pro users can run it.
+</Tip>
+
+Learn more about [configuring packages](/cli/packages/building-workflows#package-metadata-codemodyaml).
+
+## Publishing from CI/CD
+
+The easiest way to set up CI/CD publishing is to use the [migrations template repository](https://github.com/codemod/migrations-template).
+
+<Card title="Migrations Template" icon="github" href="https://github.com/codemod/migrations-template">
+  The migrations template repository includes a preconfigured GitHub Action workflow for automatic publishing.
+</Card>
+
+This GitHub template includes:
+
+- Preconfigured GitHub Action workflow for automatic publishing
+- Setup instructions for GitHub App installation and API key configuration
+- Repository structure optimized for codemod packages
+
+After setup, codemods are automatically published to the registry when PRs are merged. See the template repository for complete setup instructions.
+
+## Using Packages in Workflows
+
+You can reference registry packages in your workflow definitions:
+
+```yaml workflow.yaml
+steps:
+  - name: Run registry codemod
+    codemod:
+      source: "@nodejs/create-require-from-path@1.1.0"
+```
+
+Learn more about [using Codemod packages in workflows](/cli/packages/building-workflows#codemod-registry-step).
+
+## Best Practices
+
+### Package Naming
+
+- Use descriptive names that indicate the codemod's purpose
+- Follow the naming convention: `/^[a-z0-9-_/]+$/`
+- Use scoped names for better discoverability (e.g., `@scope/package-name`)
+- Include framework/library names when relevant
+
+### Versioning
+
+- Follow [semantic versioning](https://semver.org/) (MAJOR.MINOR.PATCH)
+- Increment major version for breaking changes
+- Increment minor version for new features
+- Increment patch version for bug fixes
+
+### Documentation
+
+- Write clear descriptions in `codemod.yaml`
+- Add relevant keywords for discoverability
+- Include examples in your README
+- Document any parameters or configuration options
+
+### Testing
+
+- Test your codemod thoroughly before publishing
+- Use [jssg testing](/jssg/testing) for automated test cases
+- Validate your workflow with `npx codemod workflow validate`
+- Test with real codebases before publishing
+
+### Access Control
+
+- Use `public` access for community-shared codemods
+- Use `private` or `org` visibility for internal packages
+- Consider `pro` access for premium codemods requiring Pro plan features
```

Comment:
this does not make sense. Pro codemods is something that we publish or authorize to publish. we can say:

Pro codemods are reserved for Codemod and their partners. To learn more contact the Codemod team. cc @mohab-sameh cc @mohebifar